### DISCUSSION TITLE
ProtonVPN Multihop/Secure core does not work

### DISCUSSION THREAD

**Discussion by Z0pyrus** [qdm12/gluetun#781] [Ideas]
When I try to set the vars SERVER_HOSTNAME (with one of the given domains) or SERVER_NAME (with the name, for example PL-CH#1) it will either way not work or connect to a random server.
If I use "SERVER_HOSTNAME=pl-ch.protonvpn.com" it say's I need to set a Name (Like PL-CH#1). When I do that it Say's i need to pass an domain.
Using the "SERVER_NAME=PL-CH#1" does nothing and will connect me to a random server.

**Comment by qdm12** [comment]
`SERVER_HOSTNAME` is the environment variable to use. If you check out https://github.com/qdm12/gluetun/wiki/Protonvpn-servers `pl-ch.protonvpn.com` does not seem to be listed there so that's why it complains. I just updated the servers information from Protonvpn's web API and there is no such hostname still now.

`SERVER_NAME` is actually not meant for ProtonVPN, that's why it's ignored; I removed it from the Wiki page.

**Reply by Z0pyrus** [reply]
It was just a example because no matter what i was typing it. I think I used "ch-us-01e.protonvpn.com"

**Comment by qdm12** [comment]
@Z0pyrus actually this is a bug introduced recently. It's fixed in 6f5a78c22b4293c3d28686d5de1eae3d2ebfd2c7

It was mixing up server hostnames and server names. You can also use `SERVER_NAME` as you did.

**Reply by Z0pyrus** [reply]
Nice. Thanks a lot!

**Comment by Z0pyrus** [comment]
So, can I use the docker image or do I need to build it from github to use the changes?

**Reply by qdm12** [reply]
Just pull the latest image `docker pull qmcgaw/gluetun`

**Comment by Z0pyrus** [comment]
Ok, it does kinda work. If I enter the hostname CH-FR#1 I will get out in FR but over the hostname "lxc-fr-12.protonvpn.com" which is Luxenbourg instead of Switzerland. In the end I get connected to France, but not over Switzerland

**Reply by qdm12** [reply]
Are you sure? From the servers.json config I see

```json
      {
        "country": "France",
        "region": "",
        "city": "",
        "name": "CH-FR#1",
        "hostname": "ch-fr-01a.protonvpn.com",
        "entry_ip": "185.159.157.90",
        "exit_ip": [
          "45.89.174.70"
        ]
      },
```

Is it Openvpn showing `lxc-fr-12.protonvpn.com`? Maybe they use the same certificate for both hops lux->fr and zurich->fr 🤔

**Reply by Z0pyrus** [reply]
That could be it. I tried some other secure connections/hops in Europe and it's basically all lxc-........protonvpn.com

**Reply by qdm12** [reply]
Probably just ask them 😉
